channel.go

```diff
@@ -230,6 +230,11 @@ func (ch *Channel) sendOpen(msg message) (err error) {
 			size = len(body)
 		}
 
+		// If the channel is closed, use Channel.sendClosed()
+		if atomic.LoadInt32(&ch.closed) == 1 {
+			return ch.sendClosed(msg)
+		}
+
 		if err = ch.connection.send(&methodFrame{
 			ChannelId: ch.id,
 			Method:    content,
```

Comment:
Isn't this vulnerable to a race condition?
If the channel gets closed after the `ch.closed` check send will be called on the closed channel

Reply:
It's less racy than before!

Reply:
Good point, let me re-visit this.

Reply:
I think my changes are correct, based on this comment -
https://github.com/rabbitmq/amqp091-go/blob/main/channel.go#L108-L110

Basically, by checking `ch.closed` at the last possible time in these two places I've narrowed the window where we could be sending the wrong data to RabbitMQ.